mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearchIndex.java

```diff
@@ -50,8 +49,32 @@
     /**
      * Return all uids of all {@link Mailbox}'s the current user has access to which match the {@link SearchQuery}
      */
-    Map<MailboxId, Collection<MessageUid>> search(MailboxSession session, MultimailboxesSearchQuery searchQuery) throws MailboxException;
+    List<MessageId> search(MailboxSession session, MultimailboxesSearchQuery searchQuery, long limit) throws MailboxException;
 
     EnumSet<MailboxManager.SearchCapabilities> getSupportedCapabilities();
 
+    class SearchResult {
```

Comment:
In later commits I made the MessageId optional (as some mailbox implementation might not support it, and in this case search should adapt), thus I can not use ComposedMessageId